### ISSUE TITLE
New recipes from other awesome cookbook

### ISSUE THREAD

**Issue by jiacai2050** [zigcc/zig-cookbook#44]
Some examples I find:
- https://github.com/dabeaz/python-cookbook
- https://golangcookbook.com/

**Comment by horochx** [comment]
I will add an example in *15. Text Processing* similar to the [Turning an Array into a Sentence](https://golangcookbook.com/chapters/strings/sentence/) and [Processing a String One Word or Character at a Time](https://golangcookbook.com/chapters/strings/processing/) in the Go Cookbook.

**Comment by jiacai2050** [comment]
@horochx Those two example seems tool simple to me, I don't know if it's necessary to add them.

**Comment by rakshasha-medhi** [comment]
@jiacai2050 I'm currently learning Zig and I got quite a collection of Zig code snippets...however how to know if something is too simple? I mean I've been struggling to find reference to learn Zig, hence I made the snippet collection...and I think this Zig cookbook is gonna be amazingly helpful for Zig newcomers.

For example i see there's Generate Random Numbers in the cookbook already, but it's using crypto.random. I have snippet that uses rand.DefaultPrng:

```
const RndGen = std.rand.DefaultPrng;
const tMilli = std.time.milliTimestamp;
const PageAlloc = std.heap.page_allocator;

fn gen_randNum(minVal: u32, maxVal: u32, arraySize: u32) ![]u32 {
    var seed = tMilli();
    var rnd = RndGen.init(@intCast(seed));
    var randNums = try PageAlloc.alloc(u32, arraySize);

    for (randNums) |*num| {
        num.* = rnd.random().intRangeAtMost(u32, minVal, maxVal);
    }
    return randNums;
}
```

I also got plenty of Sorting and Search Algorithms written in Zig.

To be honest I have been looking for an avenue to contribute to Zig learning for newcomers...so I would love to be involved in the Zig Cookbook. Cheers!
